api/v1alpha1/envoyproxy_metric_types.go

```diff
@@ -32,6 +32,11 @@ type ProxyMetrics struct {
 	// EnablePerEndpointStats enables per endpoint envoy stats metrics.
 	// Please use with caution.
 	EnablePerEndpointStats bool `json:"enablePerEndpointStats,omitempty"`
+
+	// EnableRequestResponseSizesStats enables publishing of histograms tracking header and body sizes of requests and responses.
+	//
+	// +optional
+	EnableRequestResponseSizesStats bool `json:"enableRequestResponseSizesStats,omitempty"`
```

Comment:
Nit: suggest using *bool for optional field.  EnablePerEndpointStats can also be changed to pointer. It's back-compatible.

Additionally (not in this PR), We should consider standardizing the optional bool in the APIs. While most of them are  already using pointer,  there are several that are not.

Reply:
We can make `EnableRequestResponseSizesStats` as a bool pointer type, and raise another issue to discuss if we standarlize the other bool vars in APIs

Reply:
I also thought `EnablePerEndpointStats` should be `* bool`, wrote it as `bool` to maintain consistency.

Now i have changed `EnableVirtualHostStats`, `EnablePerEndpointStats` and `EnableRequestResponseSizesStats` to `*bool` in `ProxyMetrics`.  `EnableVirtualHostStats` from usages looked like optional only to me, let me know if this needs to be changed back.

Ideally `ir.Metrics` could also have `*bool` but not changing here as i felt could result in unnecessary refactor for this issue, let me know if that needs to be changed.

Also wanted to add a function like
```
func boolValue(b *bool) bool {
    if b == nil {
        return false
    }
    return *b
}
```
ideally to extract bool, but wasn't sure where should i add it, (in same file or is there common utils used across project), so for now haven't added this.